### PR TITLE
575 sighting post remove unit tests

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Changed existing tests that use Sighting POST to use asset_group POST
- Fixed a couplpe of bugs in the code too.

---

Application changes:
locationId is now mandatory at the Sighting level
Commit response Sighting contains fuller information, including encounters, it was just the sighting guid.

Test changes Logic:
Individuals require an Encounter, Encounters require a Sighting, Sightings require an asset group to be committed. 

Utils refactored and hopefully simplified quite a bit more. 
Asset Group create returns the response message.
Sighting create (two variants, one large, one small), extracts the guids from this (transaction, asset_group, assets, asset_group_sighting) then  commits the asset group sighting and extracts the sighting guid and the encounter guid(s) from the response. 
The return value is a dictionary of guids. 
Individual creation uses this and adds an individual on one of the encouters. It is still possible to create an individual on an existing encounter, but this is the most commonly used API. 
Again the return value is the dictionary of the guids but including the individual. 

Then each test extracts the guids it needs and as necessary, the objects. For the most part it's only the guids that are required.

### CODE REVIEW

File: tests/modules/asset_groups/resources/utils.py
Comment (naknomum) [inline tests/modules/asset_groups/resources/utils.py:632]: there it goes!  :wave:

File: app/modules/asset_groups/metadata.py
Comment (hwindsor) [inline app/modules/asset_groups/metadata.py:207]: It was mandatory in the Sighting create so I made this mandatory here too.

File: app/modules/asset_groups/resources.py
Comment (hwindsor) [inline app/modules/asset_groups/resources.py:483]: Previouusly the response just contained the sighting guid. This seemed to be a far more sensible option (and made the tests easier too)

File: tests/modules/social_groups/resources/test_social_group.py
Comment (colinwkingen) [inline tests/modules/social_groups/resources/test_social_group.py:18]: Good to know

**Review by colinwkingen — APPROVED** [review]
LGTM, will wait for parallel review from Jon before merge

**Review by naknomum — APPROVED** [review]
looks good.  lotta work, thanks howard.

**Pull request merged**
